tests/src/3d/testqgs3drendering.cpp

```diff
@@ -511,5 +514,66 @@ void TestQgs3DRendering::testAnimationExport()
   QVERIFY( QFileInfo( QDir( dir ).filePath( QStringLiteral( "test3danimation001.png" ) ) ).exists() );
 }
 
+void TestQgs3DRendering::testBillboardRendering()
+{
+  QgsRectangle fullExtent( 0, 0, 1000, 1000 );
+
+  std::unique_ptr<QgsVectorLayer> layerPointsZ( new QgsVectorLayer( "PointZ?crs=EPSG:27700", "points Z", "memory" ) );
+
+  QgsPoint *p1 = new QgsPoint( 0, 0, 50 );
+  QgsPoint *p2 = new QgsPoint( 0, 1000, 100 );
+  QgsPoint *p3 = new QgsPoint( 1000, 1000, 200 );
+
+  QgsFeature f1( layerPointsZ->fields() );
+  QgsFeature f2( layerPointsZ->fields() );
+  QgsFeature f3( layerPointsZ->fields() );
+
+  f1.setGeometry( QgsGeometry( p1 ) );
+  f2.setGeometry( QgsGeometry( p2 ) );
+  f3.setGeometry( QgsGeometry( p3 ) );
+
+  QgsFeatureList featureList;
+  featureList << f1 << f2 << f3;
+  layerPointsZ->dataProvider()->addFeatures( featureList );
+
+  QgsMarkerSymbol *markerSymbol = static_cast<QgsMarkerSymbol *>( QgsSymbol::defaultSymbol( QgsWkbTypes::PointGeometry ) );
+  markerSymbol->setColor( QColor( 255, 0, 0 ) );
+  QgsPoint3DSymbol *point3DSymbol = new QgsPoint3DSymbol( markerSymbol );
+  point3DSymbol->setShape( QgsPoint3DSymbol::Billboard );
+
+  layerPointsZ->setRenderer3D( new QgsVectorLayer3DRenderer( point3DSymbol ) );
+
+  Qgs3DMapSettings *map = new Qgs3DMapSettings;
+  map->setCrs( mProject->crs() );
+  map->setOrigin( QgsVector3D( fullExtent.center().x(), fullExtent.center().y(), 0 ) );
+  map->setLayers( QList<QgsMapLayer *>() << layerPointsZ.get() );
```

Comment:
can we have different color for the symbol outline from the scene background color? Now it is unclear in the rendering output what is still part of the billboard and what is already the background...